assets/src/edit-story/components/panels/document/slug/slug.js

```diff
@@ -38,24 +43,39 @@ export const MIN_MAX = {
   },
 };
 
-const BoxedTextInput = styled(TextInput)`
-  padding: 6px 6px;
-  border-radius: 4px;
-  flex-grow: 1;
-  &:focus {
-    background-color: ${({ theme }) => theme.DEPRECATED_THEME.colors.fg.white};
+const PermalinkRow = styled(Row)`
+  margin-bottom: 12px;
+
+  * input {
+    color: ${({ theme }) => theme.colors.fg.tertiary};
+
+    :active,
+    :focus,
+    .${ThemeGlobals.FOCUS_VISIBLE_SELECTOR} {
+      color: ${({ theme }) => theme.colors.fg.primary};
+    }
   }
 `;
 
+const LinkContainer = styled.div`
+  margin-bottom: 16px;
+`;
+
 function SlugPanel() {
-  const { slug, link, permalinkConfig, updateStory } = useStory(
+  const { slug: savedSlug, link, permalinkConfig, updateStory } = useStory(
     ({
       state: {
         story: { slug = '', link, permalinkConfig },
       },
       actions: { updateStory },
     }) => ({ slug, link, permalinkConfig, updateStory })
   );
+  const [slug, setSlug] = useState(savedSlug);
+
+  useEffect(() => {
+    /* Update shown slug when slug updates externally */
+    setSlug(savedSlug);
+  }, [savedSlug]);
 
   const updateSlug = useCallback(
     (value, isEditing) => {
```

Comment:
If we're using this only `onBlur` then the `isEditing` can be removed here and the `cleanForSlug` used without `isEditing` inside the `updateSlug` (since it defaults to `false`).